From b54bb00e2bf3344ff740b5a642dc1a7c70b3d76f Mon Sep 17 00:00:00 2001
From: Andrey Zhizhikin <andrey.zhizhikin@leica-geosystems.com>
Date: Sun, 26 Jan 2020 10:45:28 +0000
Subject: [PATCH] ext/wayland: fix meson build in nxp fork

- Add Wayland protocols, which are not enabled in upstream
- Add missing compilation unit, which in NXP-specific

Upstream-Status: Pending

Signed-off-by: Andrey Zhizhikin <andrey.zhizhikin@leica-geosystems.com>
---
 ext/wayland/meson.build | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/ext/wayland/meson.build b/ext/wayland/meson.build
index 3c377eefb..5ef082dac 100644
--- a/ext/wayland/meson.build
+++ b/ext/wayland/meson.build
@@ -8,6 +8,11 @@ wl_sources = [
     'wllinuxdmabuf.c'
 ]
 
+imx_wl_sources = [
+    'wlutils.c',
+]
+wl_sources += imx_wl_sources
+
 libdrm_dep = dependency('libdrm', version: '>= 2.4.55', required:get_option('wayland'))
 
 if use_wayland
@@ -21,6 +26,25 @@ if use_wayland
          'fullscreen-shell-unstable-v1-protocol.c', 'fullscreen-shell-unstable-v1-client-protocol.h'],
         ['/stable/xdg-shell/xdg-shell.xml', 'xdg-shell-protocol.c', 'xdg-shell-client-protocol.h'],
     ]
+
+    imx_protocol_defs = [
+        [
+         '/unstable/alpha-compositing/alpha-compositing-unstable-v1.xml',
+         'alpha-compositing-unstable-v1-protocol.c',
+         'alpha-compositing-unstable-v1-client-protocol.h'
+        ],
+        ['/unstable/hdr10-metadata/hdr10-metadata-unstable-v1.xml',
+         'hdr10-metadata-unstable-v1-protocol.c',
+         'hdr10-metadata-unstable-v1-client-protocol.h'
+        ],
+        [
+         '/unstable/linux-explicit-synchronization/linux-explicit-synchronization-unstable-v1.xml',
+         'linux-explicit-synchronization-unstable-v1-protocol.c',
+         'linux-explicit-synchronization-unstable-v1-client-protocol.h'
+        ],
+    ]
+    protocol_defs += imx_protocol_defs
+
     protocols_files = []
 
     foreach protodef: protocol_defs
-- 
2.20.1

